library/waku_thread/waku_thread.nim

```diff
@@ -83,20 +89,24 @@ proc createWakuThread*(): Result[void, string] =
 proc stopWakuNodeThread*() =
   running.store(false)
   joinThread(ctx.thread)
-
-  ctx.reqChannel.close()
-  ctx.respChannel.close()
-
   freeShared(ctx)
 
-proc sendRequestToWakuThread*(req: InterThreadRequest): Result[string, string] =
+proc sendRequestToWakuThread*(reqType: RequestType,
+                              reqContent: pointer): Result[string, string] =
 
-  ctx.reqChannel.send(req)
+  let req = InterThreadRequest.createShared(reqType, reqContent)
 
-  var resp = ctx.respChannel.tryRecv()
-  while resp[0] == false:
-    resp = ctx.respChannel.tryRecv()
-    os.sleep(1)
+  ## Sending the request
+  let sentOk = ctx.reqChannel.trySend(req)
+  if not sentOk:
+    return err("Couldn't send a request to the waku thread: " & $req[])
 
-  return resp[1]
+  ## Waiting for the response
+  var response: ptr InterThreadResponse
+  var recvOk = ctx.respChannel.tryRecv(response)
+  while recvOk == false:
```

Comment:
instead of this loop, a signal can be used here too

Reply:
Okay!, we'll apply the `signal` enhancements in a future PR.